data.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+from torch.utils.data import Dataset, DataLoader
+from torchvision import datasets
+import uproot
+from pathlib import Path
+import json
+import time # used only for runtime testing
+from debug.fastloader import FastTensorDataLoader
+
+##
+# This file contains the Data Manager and Data Loader modules (Using PyTorch in-built dataloader)
+##
+
+class DataManager():
+    def __init__(self, input_file="input_files/input.json"):
+        with open(input_file) as f:
+            args = json.load(f)
+        self.args = args
+        if(args["dir_name_processing"]):
+            self.args["particles"] = list(map(self.string_processing, self.args["particles"]))
+
+        # Resolve relative paths to absolute paths
+        self.args["input_paths"] = list(map( (lambda x : Path(x).resolve()), self.args["input_paths"]))
+        self.args["output_paths"] = list(map( (lambda x : Path(x).resolve()), self.args["output_paths"]))
+
+    def string_processing(self, string):
+        return( string + "/fCoordinates/fCoordinates." )
+
+class CustomDataset(Dataset):
+    def __init__(self, manager):
+        self.manager = manager
+
+    def __len__(self):
+        file = uproot.open(self.manager.args["output_paths"][0])
+        len = file[self.manager.args["tree"]].arrays(self.manager.args["weights"], library="pd").shape[0]
+        file.close()
+        return len
+
+    def getitemhelper(self, idx):
```

Comment:
```suggestion
    def get_item_helper(self, idx):
```

It is generally a good idea to use Pythonic naming schemes. In Python classes are CameCase and everything else is snake_case. I'd apply snake_case to all functions and methods.

Reply:
Hello, while I can change "getitemhelper" to "get_item_helper" the other functions like "__getitem__" (for which getitemhelper is a helper method) are inherited from the base Dataset class and cannot be changed.

Reply:
Yeah not much you can do there. In that case it is probably better to have a more consistent API.